src/instrumentation/instrumentation_validator.go

```diff
@@ -69,6 +77,22 @@ func (r *InstrumentationValidator) validate(inst *v1alpha2.Instrumentation) (adm
 	if inst.Spec.Agent.IsEmpty() {
 		return nil, fmt.Errorf("instrumentation %q agent is empty", inst.Name)
 	}
+	if len(inst.Spec.HealthAgent.Env) > 0 && inst.Spec.HealthAgent.Image != "" {
```

Comment:
just noting what you found earlier

```suggestion
	if len(inst.Spec.HealthAgent.Env) > 0 && inst.Spec.HealthAgent.Image == "" {
```

Reply:
fixed